test/uk/gov/hmrc/transitmovements/services/ObjectStoreServiceSpec.scala

```diff
@@ -0,0 +1,110 @@
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.http.Status.INTERNAL_SERVER_ERROR
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.http.UpstreamErrorResponse
+import uk.gov.hmrc.objectstore.client.Md5Hash
+import uk.gov.hmrc.objectstore.client.Object
+import uk.gov.hmrc.objectstore.client.ObjectMetadata
+import uk.gov.hmrc.objectstore.client.Path
+import uk.gov.hmrc.objectstore.client.Path.File
+import uk.gov.hmrc.objectstore.client.play.PlayObjectStoreClient
+import uk.gov.hmrc.transitmovements.base.StreamTestHelpers
+import uk.gov.hmrc.transitmovements.base.TestActorSystem
+import uk.gov.hmrc.transitmovements.generators.ModelGenerators
+import uk.gov.hmrc.transitmovements.models.ObjectStoreURI
+import uk.gov.hmrc.transitmovements.services.errors.ObjectStoreError
+
+import java.time.Instant
+import java.util.UUID.randomUUID
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class ObjectStoreServiceSpec
+    extends AnyFreeSpec
+    with Matchers
+    with MockitoSugar
+    with TestActorSystem
+    with StreamTestHelpers
+    with BeforeAndAfterEach
+    with ModelGenerators
+    with ScalaCheckDrivenPropertyChecks {
+
+  implicit val hc: HeaderCarrier                           = HeaderCarrier()
+  private val mockObjectStoreClient: PlayObjectStoreClient = mock[PlayObjectStoreClient]
+
+  override def beforeEach: Unit =
+    reset(mockObjectStoreClient)
+
+  "Object Store service" - {
+
+    "should return the contents of a file" in {
+      val filePath =
+        Path.Directory(s"common-transit-convention-traders/movements/${arbitraryMovementId.arbitrary.sample.get}").file(randomUUID.toString).asUri
+      val metadata    = ObjectMetadata("", 0, Md5Hash(""), Instant.now(), Map.empty[String, String])
+      val content     = "content"
+      val fileContent = Option[Object[Source[ByteString, NotUsed]]](Object.apply(File(filePath), Source.single(ByteString(content)), metadata))
+      when(mockObjectStoreClient.getObject[Source[ByteString, NotUsed]](any[File](), any())(any(), any())).thenReturn(Future.successful(fileContent))
+      val service = new ObjectStoreServiceImpl(mockObjectStoreClient)
+      val result  = service.getObjectStoreFile(ObjectStoreURI(filePath))
+      whenReady(result.value) {
+        r =>
+          r.isRight mustBe true
+          r.toOption.get mustBe fileContent.get.content
+
+      }
+    }
+
+    "should return an error when the file is not found on path" in {
+      when(mockObjectStoreClient.getObject(any[File](), any())(any(), any())).thenReturn(Future.successful(None))
+      val service = new ObjectStoreServiceImpl(mockObjectStoreClient)
+      val result  = service.getObjectStoreFile(ObjectStoreURI("abc/movement/abc.xml"))
+      whenReady(result.value) {
+        r =>
+          r.isLeft mustBe true
+          r.left.getOrElse(ObjectStoreError.FileNotFound) mustBe a[ObjectStoreError.FileNotFound]
+      }
```

Comment:
Maybe use pattern matching instead? Something like...

```scala
whenReady(result.value) {
    case Left(_: ObjectStoreError.FileNotFound) => succeed
    case x => fail(s"Expected Left(ObjectStoreError.FileNotFound), instead got $x")
}
```

Just looking for something that's more readable (I realise this might have been taken from previous tests (probably mine), but we should look to improve as we learn more about what we can do with Scala and such.)